Fixing as-needed issues

http://bugs.gentoo.org/show_bug.cgi?id=317905

Patch written by Kacper Kowalik <xarthisius.kk@gmail.com>

--- jb
+++ jb
@@ -89,10 +89,11 @@
 
 		mkdir -p build/jbsrc || exit 1
 
-		if ! gobject_cflags_libs=`pkg-config --cflags --libs 'gobject-2.0 >= 2.8.0' 2>/dev/null`; then
+		if ! gobject_cflags=`pkg-config --cflags 'gobject-2.0 >= 2.8.0' 2>/dev/null`; then
 			echo "ERROR: the GLib library was not found" >&2
 			exit 1
 		fi
+		gobject_libs=`pkg-config --libs 'gobject-2.0 >= 2.8.0' 2>/dev/null`
 
 		sed -e 's|^#jb_include ["<]\(.*\)[">]|#include "\1.h"|' \
 			jbsrc/jb.c > build/jbsrc/jb.c || exit 1
@@ -103,14 +104,15 @@
 			-DG_LOG_DOMAIN="\"jb\"" \
 			-DJB_SOURCES="\"$jb_c_sources $jb_h_sources\"" \
 			-DJB_PACKAGE_SOURCES="\"$jb_c_package_sources $jb_h_package_sources\"" \
-			$gobject_cflags_libs \
+			$gobject_cflags \
 			$jb_cflags \
 			$jb_cppflags \
 			$jb_ldflags \
 			build/jbsrc/jb.c \
 			$jb_c_sources \
 			$jb_c_package_sources \
-			$jb_libs; then
+			$jb_libs \
+			$gobject_libs; then
 			echo "ERROR: cannot build jb" >&2
 			exit 1
 		fi
